DatadogRUM/Tests/RUMMonitor/Scopes/RUMViewScopeTests.swift

```diff
@@ -3390,4 +3390,126 @@ class RUMViewScopeTests: XCTestCase {
         XCTAssertEqual(events[1].session.hasReplay, true)
         XCTAssertEqual(events[2].session.hasReplay, true)
     }
+
+    // MARK: - View Attributes
+    @available(iOS 13.0, tvOS 13.0, *)
+    @MainActor
+    func testAccessibilityAttributesInViewEvents() throws {
+        // Given
+        let mockAccessibilityState = Accessibility(
+            textSize: "medium",
+            screenReaderEnabled: false,
+            boldTextEnabled: true,
+            reduceTransparencyEnabled: nil,
+            reduceMotionEnabled: nil,
+            buttonShapesEnabled: nil,
+            invertColorsEnabled: nil,
+            increaseContrastEnabled: nil,
+            assistiveSwitchEnabled: nil,
+            assistiveTouchEnabled: nil,
+            videoAutoplayEnabled: nil,
+            closedCaptioningEnabled: nil,
+            monoAudioEnabled: nil,
+            shakeToUndoEnabled: nil,
+            reducedAnimationsEnabled: nil,
+            shouldDifferentiateWithoutColor: nil,
+            grayscaleEnabled: nil,
+            singleAppModeEnabled: nil,
+            onOffSwitchLabelsEnabled: nil,
+            speakScreenEnabled: nil,
+            speakSelectionEnabled: nil,
+            rtlEnabled: nil
+        )
+
+        let mockAccessibilityReader = AccessibilityReaderMock(state: mockAccessibilityState)
+
+        let mockParent = RUMContextProviderMock()
+        let testContext = context
+
+        let dependencies = RUMScopeDependencies.mockWith(
+            accessibilityReader: mockAccessibilityReader
+        )
+
+        let scope = RUMViewScope(
+            isInitialView: false,
+            parent: mockParent,
+            dependencies: dependencies,
+            identity: .mockViewIdentifier(),
+            path: "UIViewController",
+            name: "MyView",
+            customTimings: [:],
+            startTime: .mockDecember15th2019At10AMUTC(),
+            serverTimeOffset: .zero,
+            interactionToNextViewMetric: INVMetricMock()
+        )
+
+        let readerInitExpectation = XCTestExpectation(description: "AccessibilityReader initialized")
+        DispatchQueue.main.async {
+            DispatchQueue.main.async {
+                readerInitExpectation.fulfill()
+            }
+        }
+        wait(for: [readerInitExpectation], timeout: 2.0)
```

Comment:
question/ Do we actually need this 🤔? Since we're using a mock `AccessibilityReader` that returns state synchronously, do we still need this extra asynchronous waiting? All we want to test is whether `RUMViewScope` reads the accessibility attributes correctly from the reader. While the default reader evaluates the state asynchronously, our mock returns it immediately.